lib/conduit.ml

```diff
@@ -52,7 +52,7 @@ module type S = sig
   val send : flow -> output -> (int, [> `Msg of string ]) result s
   val close : flow -> (unit, [> `Msg of string ]) result s
 
-  type 'edn resolver = [ `host ] Domain_name.t -> ('edn option) s
+  type 'edn resolver = [ `raw ] Domain_name.t -> ('edn option) s
```

Comment:
I've a question about this commit: did you encounter any non-host domain names that are used by conduit clients? IMHO it would be nice to use `` [ `host  ] Domain_name.t``.

Reply:
Yes, I'm not fixed about that and before, I used `[ host ]`. However, it seems that into `ocaml-cohttp`, or precisely into `Uri.t`, we can pass an IPv4 such as `http://0.0.0.0/` which does not contain a valid `[ host ] Domain_name.t`.

A possible solution is to update the `cohttp`'s test suite to pass `localhost` instead.

Reply:
if an Uri can be like this, it is also special in conduit/tuyau, since there's no need to resolve it via DNS -- i.e. that step can be saved (and when doing TLS, it should also verify the IP address in the certificate rather than the hostname [this is not implemented IIRC])....

but i think it is worth keeping host and handling ip addresses in a uri.t special (of course this may be a separate PR)